internal/resource/nvml-device.go

```diff
@@ -86,3 +90,28 @@ func (d nvmlDevice) GetTotalMemoryMB() (uint64, error) {
 	}
 	return info.Total / (1024 * 1024), nil
 }
+
+func (d nvmlDevice) GetClass() (string, error) {
+	info, retVal := d.Device.GetPciInfo()
+	if retVal != nvml.SUCCESS {
+		return "", retVal
+	}
+	var bytes []byte
+	for _, char := range info.BusId {
+		if char == 0 {
+			break
+		}
+		bytes = append(bytes, byte(char))
+	}
+	pciID := strings.ToLower(strings.TrimPrefix(string(bytes), "0000"))
+	return resolvePCIAddressToClass(pciID)
+}
+
+func resolvePCIAddressToClass(addr string) (string, error) {
+	class, err := os.ReadFile(fmt.Sprintf("/sys/bus/pci/devices/%s/class", addr))
+	if err != nil {
+		klog.Errorf("Error getting gpu class: %v", err)
+		return "unknown", fmt.Errorf("Error getting gpu class: %v", err)
+	}
+	return strings.TrimSpace(string(class)), nil
```

Comment:
What about reusing the logic from `nvpci`:
```suggestion
	// TODO: The nvpci.Interface instance should be a member of 
	// nvmlDevice.
	device, err := nvpci.New().GetGPUByPciBusID(addr)
	if err != nil {
		return "", err
	}
	return fmt.Sprintf("%#06x", device.Class), nil
```

Reply:
Fixed this